extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -478,7 +479,7 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 				// since bookItems will not have them yet, check the sections->file property
 				bookItemToExpand = parentBook.bookItems.find(b => b.sections?.find(n => notebookFolders[notebookFolders.length - depthOfNotebookInBook - 1].indexOf(n.file.substring(n.file.lastIndexOf('/') + 1)) > -1));
 				// book isn't found even in the same level, break out and return.
-				if (!bookItemToExpand) {
+				if (!bookItemToExpand || expandedBookItems.includes(bookItemToExpand)) {
```

Comment:
Do you have to keep track of all the book items you've found? Or would it just be enough to instead just break out if we get to a node that doesn't have children and we haven't found the book yet?

Reply:
I think it would be better to keep looking since we can have weird book structures when it comes to livesite.

Reply:
But this isn't going to "keep looking", right? It just traverses down the path checking the folders along the way until it finds the item - so if it ever gets to the point where there aren't any children then there's nothing more for us to check (which is where it would get into the endless loop).



If you wanted to fix it so that it checks all other paths then that'd be a separate enhancement.

Reply:
What I meant by keep looking is I don't want to break out if we get to a node that doesn't have children (keep the iteration going until the depth condition is satisfied)

Reply:
Can you verify that's what actually happens? Looking at the code it seems like the state doesn't change - on 486 it increments the depth so that on 501 when it decrements it it'll actually stay in the same place. So it'll just go and repeat what it just tried.

The idea being that if there were children that were loaded it'll now check them to see if any of them were the one we're looking for. But if there's no children then isn't that where we got into the endless loop in the first place?

Reply:
I double checked and I added that increment step for multiple child nodes in the same level. But am not checking it it's the same child node that it's trying to expand, so added the expanded nodes to check that and not increment it if it's already expanded node.